src/off_chain/historical_data/cow_endpoint_surplus.py

```diff
@@ -0,0 +1,295 @@
+"""
+EBBO Historical Data Testing via block number inputs or a single settlement hash.
+Uses CoW Endpoint provided callData.
+"""
+import json
+import requests
+from fractions import Fraction
+import directory
+from configuration import solver_dict, header, get_logger
+from typing import List, Dict, Tuple, Any, Optional
+
+
+class EBBOHistoricalDataTesting:
```

Comment:
Since this class now works for both historical and live monitoring, should I rename it to something like EBBOAnalysis?

Reply:
Yeah, I would actually remove the historical part there. This is really the core of EBBO; it gets a block/block interval/hash, and it will report back the result. So, I agree we should get rename